### PR TITLE
Add npm and docker publish artifacts

### PR DESCRIPTION
**Pull request by aklenik**
Signed-off-by: Attila Klenik <a.klenik@gmail.com>

This PR prepares the repo for NPM and Docker publishing.

The bigger changes include:
1. Renaming the packages for the HL scope: `@hyperledger/caliper-X` (this will avoid any naming conflict with other public packages)
1. Adjusting references (requires, deps, package.json) for the new package names
1. Moving the adapter SDK deps into devDeps, so they can be skipped during user install (since the bind command will install them anyway)
1. Extending the CLI with a bind command (and related configuration) to bootstrap the specified platform SDK deps runtime.
1. Cleaning up the integration testing scripts and providing other utility scripts for publishing
1. Adding Dockerfile file for publishing

The smaller changes include:
1. Making the CLI parameters optional, so they can also be set from env vars (useful for Docker, and flexibility++)
1. Resolving log file path relative to the workspace and not the current working dir
1. Adding license check exception for log files

The documentation preview for the install/dev/publish workflow: https://aklenik.github.io/caliper/docs/Installing_Caliper.html

### CODE REVIEW

File: packages/caliper-fabric/package.json [hunk 1]
```diff
@@ -23,15 +23,14 @@
     "src/comm/template/report.html"
   ],
   "dependencies": {
-    "caliper-core": "0.1.0",
+    "@hyperledger/caliper-core": "0.1.0",
+    "fs-extra": "^8.0.1"
```
Comment (nklincoln) [inline packages/caliper-fabric/package.json]: we no longer use fs-extra in the caliper-fabric adaptor

File: packages/caliper-iroha/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "caliper-iroha",
+  "name": "@hyperledger/caliper-iroha",
   "version": "0.1.0",
   "repository": {
     "type": "git",
```
Comment (nklincoln) [inline packages/caliper-iroha/package.json]: within the scripts section, we can remove the install iroha deps

File: packages/caliper-sawtooth/package.json [hunk 3]
```diff
@@ -19,13 +19,13 @@
     "npm": ">=5.6.0"
   },
   "dependencies": {
-    "caliper-core": "0.1.0",
+    "@hyperledger/caliper-core": "0.1.0",
     "protocol-buffers": "^4.1.0",
-    "sawtooth-sdk": "^1.0.5",
     "cbor": "^3.0.3",
```
Comment (nklincoln) [inline packages/caliper-sawtooth/package.json]: unused dep

File: packages/caliper-sawtooth/package.json [hunk 4]
```diff
@@ -19,13 +19,13 @@
     "npm": ">=5.6.0"
   },
   "dependencies": {
-    "caliper-core": "0.1.0",
+    "@hyperledger/caliper-core": "0.1.0",
     "protocol-buffers": "^4.1.0",
```
Comment (nklincoln) [inline packages/caliper-sawtooth/package.json]: unused dependancy

File: packages/caliper-fabric/package.json (hunk 1 above)
Comment (nklincoln) [reply packages/caliper-fabric/package.json]: unused dep `fs-extra`

**Comment by aklenik** [comment]
@nklincoln Applied the changes. Plus I had to add .gitignore to every package dir, so npm publish can use it also

**Comment by aklenik** [comment]
Aaaand also fixed the gitignore license issues...

**Pull request merged**